In [1]:
import pandas as pd
import numpy as np
import scipy as sp

# Лабораторная работа №2
## Задание 2
## Кондрашин Тимофей 211-365 Вариант 60 (20.04.2023)

In [2]:
data = pd.DataFrame(
    [21.3, 26.5, 39.5, 17.5, 27.5, 38.2, 29.0, 30.1, 27.5, 32.7, 19.4, 22.2, 36.4, 37.3, 25.0, 25.7, 31.8, 24.1, 30.0,
     27.4, 27.0, 22.9, 30.0, 28.2, 23.9, 18.1, 18.9, 26.9, 27.8, 16.5, 32.6, 33.4, 26.3, 20.4, 24.9, 28.7, 41.5, 22.4,
     20.5, 35.3, 14.1, 34.4, 35.0, 23.7, 25.2, 31.5, 37.2, 20.7, 33.2, 28.0, 22.0, 31.7, 23.4, 30.8, 35.0, 31.4, 28.6,
     13.9, 34.6, 31.4, 30.8, 19.4, 29.3, 27.3, 37.6, 24.6, 32.4, 25.4, 27.7, 28.6, 33.7, 26.9, 29.4, 28.2, 32.9, 29.5,
     20.5, 26.5, 32.2, 32.9, 26.4, 28.9, 29.2, 32.6, 23.2, 31.7, 17.8, 24.9, 28.5, 35.6, 27.2, 29.1, 29.9, 27.2, 31.2,
     27.8, 34.1, 29.0, 26.5, 24.8])
data.T

0     1     2     3     4     5     6     7     8     9   ...    90   
0  21.3  26.5  39.5  17.5  27.5  38.2  29.0  30.1  27.5  32.7  ...  27.2  \

     91    92    93    94    95    96    97    98    99  
0  29.1  29.9  27.2  31.2  27.8  34.1  29.0  26.5  24.8  

[1 rows x 100 columns]

In [3]:
data.describe()

0
count  100.000000
mean    28.025000
std      5.594954
min     13.900000
25%     24.875000
50%     28.200000
75%     31.725000
max     41.500000

### Всего элементов ряда

In [4]:
data_len = len(data)
data_len

100

### Выборочное среднее

$$\overline x = \frac1n \sum^n_{i=1} x_i$$

In [5]:
mean = np.mean(data)
mean

28.025

### Выборочная дисперсия

$$s^2_x = \frac{1}{n-1}\sum^n_{i=1} (x_i - \overline x)^2$$

In [6]:
var = np.var(data)
var

0    30.990475
dtype: float64

### Среднеквадратическое отклонение

$$s = \sqrt{s^2}$$

In [7]:
std = np.std(data)
std

0    5.566909
dtype: float64

### $t$-критерий Стьюдента

$$t = \frac{\overline x - \mu}{\frac{s}{\sqrt{n}}}$$

In [8]:
gamma = 0.99
t_value = sp.stats.t.ppf(gamma, data_len-1)
t_value

2.3646058614359737

### Доверительный интервал для математического ожидания при доверительной вероятности $\gamma$

#### Z-оценка

$$z={x-{\bar {X}} \over S_{x}}$$

In [9]:
sp.stats.norm.ppf(.99)

2.3263478740408408

#### Доверительный интервал для математического ожидания через $z$ оценку

$$\left(\overline x - z \cdot \frac{s}{\sqrt{n}}, \overline x + z \cdot \frac{s}{\sqrt{n}}\right)$$

In [10]:
def confidence_interval(data, gamma):
    mean = np.mean(data)
    std = np.std(data)
    n = len(data)
    z = sp.stats.norm.ppf(.99)
    return (mean - z * std / np.sqrt(n), mean + z * std / np.sqrt(n))

confidence_interval(data, 0.99)

(0    26.729943
 dtype: float64,
 0    29.320057
 dtype: float64)

#### Доверительный интервал для математического ожидания через $t$-критерий Стьюдента

$$\left(\overline x - t \cdot \frac{s}{\sqrt{n}}, \overline x + t \cdot \frac{s}{\sqrt{n}}\right)$$

In [11]:
def confidence_interval_t(data, gamma):
    mean = np.mean(data)
    std = np.std(data)
    n = data_len
    t = t_value
    return (mean - t * std / np.sqrt(n), mean + t * std / np.sqrt(n))

confidence_interval_t(data, 0.99)

(0    26.708645
 dtype: float64,
 0    29.341355
 dtype: float64)

#### Вывод

Таким образом с вероятностью 0,99 можно утверждать, что математическое ожидание лежит в интервале (26.7; 29.3)